src/refract/VisitorUtils.cc

```diff
@@ -66,3 +66,22 @@ MemberElement* refract::FindMemberByKey(const ObjectElement& e, const std::strin
 
     return static_cast<MemberElement*>(it->get());
 }
+
+bool refract::IsLiteral(const IElement& e)
+{
+    TypeQueryVisitor query;
+    VisitBy(e, query);
+    auto elementType = query.get();
+
+    switch (elementType) {
+        case TypeQueryVisitor::String:
+        case TypeQueryVisitor::Number:
+        case TypeQueryVisitor::Boolean:
+            if (!e.empty())
+                return true;
+        default:
+            return HasTypeAttribute(e, "fixed");
+            ;
+    };
```

Comment:
Please make an if condition out of this switch.

Reply:
Also, what about `NullElement`?

Reply:
I mean, the function is called `IsLiteral`. I think `IsLiteral` should behave the same for all four primitive Elements.
There is no such concept as `fixed` on `Null Element`, because there is just one value in the Null universe. For that reason I'd just always use `{"element":"null"}` for Null Element specifically.

Reply:
Oh you mean, null should not be fixed under any condition. Agree. Will fix